spec/index.md

```diff
@@ -1806,6 +1806,17 @@ Not including a setting in the `settings` record means not changing that field.
 
 The optional `sender_canister_version` parameter can contain the caller's canister version. If provided, its value must be equal to `ic0.canister_version`.
 
+
+### IC method `upload_chunk` {#ic-upload_chunk}
```

Comment:
We should limit the maximum number of chunks that can be stored (it shouldn't be necessary to have more than 5 chunks stored at any time) as otherwise we could make the management canister API vulnerable from the security perspective.

Reply:
Agree, but I think the limit will be more than 5. We currently allow Wasms to be 30MB uncompressed, so if we want to allow them to installed using chunked uploading without being compressed and choose 512KB for the chunk size it'll be more like ~100 chunks as the minimum.

But user's might also want to store the code for multiple canisters in a single canister's Wasm store, so I think the limit should be closer to 1GB.

Reply:
We'll start with some conservative choice, use 50MB for chunked storage and the slowly increase it.